packaging/language/maven_artifact.py

```diff
@@ -257,7 +257,7 @@ def _request(self, url, failmsg, f):
         self.module.params['http_agent'] = self.module.params.get('user_agent', None)
 
         response, info = fetch_url(self.module, url_to_use, timeout=req_timeout)
-        if info['status'] != 200:
+        if info['status'] != 200 or info['msg'][:2] != 'OK':
```

Comment:
I understand now, its because `None != 200` evaluates to false and you're relying on _or_ short circuiting. Thank you for pointing out my error.

As a pure question only (not suggesting you change anything) would the intent be clearer and more obvious by reversing the result as in:
```python
if info['status'] == 200 or info['msg'][:2] == 'OK':
    return f(response)
else:
    raise ValueError(failmsg + " because of " + info['msg'] + "for URL " + url_to_use)
```
Thanks again.